packages/front-end/components/Experiment/CustomFieldDisplay.tsx

```diff
@@ -0,0 +1,130 @@
+  const [editModal, setEditModal] = useState(false);
+  const customFields = useCustomFields();
+  const customFieldsMap = new Map();
+  const defaultFields: CustomExperimentField = {};
+  if (customFields && customFields.length) {
+    customFields.map((v) => {
+      defaultFields[v.id] =
+        v.type === "boolean"
+          ? JSON.stringify("")
+          : v.type === "multiselect"
+          ? JSON.stringify([])
+          : "";
+      customFieldsMap.set(v.id, v);
+    });
+  }
+
+  const { hasCommercialFeature } = useUser();
+  const hasCustomFieldAccess = hasCommercialFeature("custom-exp-metadata");
+  const form = useForm<Partial<ExperimentInterfaceStringDates>>({
+    defaultValues: {
+      customFields: experiment?.customFields || defaultFields,
+    },
+  });
+  const { apiCall } = useAuth();
+
+  if (customFields && customFields.length) {
+    return (
+      <>
+        {editModal && (
+          <Modal
+            header={"Edit Custom Fields"}
+            open={editModal}
+            close={() => {
+              setEditModal(false);
+            }}
+            size="lg"
+            submit={form.handleSubmit(async (value) => {
+              await apiCall(`/experiment/${experiment.id}`, {
+                method: "POST",
+                body: JSON.stringify({ ...value }),
+              });
+
+              if (mutate) mutate();
+            })}
+            cta="Save"
+          >
+            {hasCustomFieldAccess ? (
+              <CustomFieldInput customFields={customFields} form={form} />
+            ) : (
+              <div className="text-center">
+                <PremiumTooltip commercialFeature={"custom-exp-metadata"}>
+                  Custom fields are available as part of the enterprise plan
+                </PremiumTooltip>
+              </div>
+            )}
+          </Modal>
+        )}
+        {label && (
+          <HeaderWithEdit
+            edit={
+              canEdit &&
+              hasCustomFieldAccess &&
+              (() => {
+                setEditModal(true);
+              })
+            }
+            outerClassName="mb-3"
+          >
+            {label}
+          </HeaderWithEdit>
+        )}
+        {experiment?.customFields && (
+          <>
+            {Array.from(customFieldsMap.values()).map((v: CustomField) => {
+              // these two loops are used to make sure the order is correct with the stored order of custom fields.
+              return Object.keys(experiment.customFields).map((fid, i) => {
+                if (v.id === fid) {
+                  const f = experiment.customFields[fid];
+                  const displayValue =
+                    v.type === "multiselect" ? JSON.parse(f).join(", ") : f;
+                  if (displayValue) {
+                    if (v.type === "textarea" || v.type === "markdown") {
+                      return (
+                        <div className="mb-4" key={i}>
+                          <h4>{v.name}</h4>
```

Comment:
UI nitpick: margins and heading are pretty large. Instead of this:
<img width="350" alt="image" src="https://user-images.githubusercontent.com/7927873/219818962-fc85d41b-b1ec-4c46-aa94-f4cc74cec1ad.png">

Maybe more like this?
<img width="322" alt="image" src="https://user-images.githubusercontent.com/7927873/219819181-debe7873-1534-4a33-9b9e-6d99b558b68f.png">

Reply:
I like that